src/minidom_utils/try_only_child.rs

```diff
@@ -26,12 +26,26 @@ use minidom::Element;
 /// fail)
 pub trait TryOnlyChild {
     /// Try to get an unique child from its name and return a [Result](crate::Result)
-    fn try_only_child<'a>(&'a self, child_name: &str) -> Result<&'a Element>;
+    /// A filter can be apply on the kind of children you want to select
+    fn try_only_child_with_filter<'a, P>(&'a self, child_name: &str, filter: P) -> Result<&'a Self>
+    where
+        P: Fn(&'a Self) -> bool;
+
+    /// Try to get an unique child from its name and return a [Result](crate::Result)
+    fn try_only_child<'a>(&'a self, child_name: &str) -> Result<&'a Self> {
+        self.try_only_child_with_filter(child_name, |_| true)
+    }
 }
 
 impl TryOnlyChild for Element {
-    fn try_only_child<'a>(&'a self, child_name: &str) -> Result<&'a Element> {
-        let mut child_iterator = self.children().filter(|child| child.name() == child_name);
+    fn try_only_child_with_filter<'a, P>(&'a self, child_name: &str, filter: P) -> Result<&'a Self>
+    where
+        P: Fn(&'a Self) -> bool,
+    {
+        let mut child_iterator = self
+            .children()
+            .filter(|child| child.name() == child_name)
```

Comment:
I should also be able to do something like
```
.filter(|child| child.name() == child_name && filter(*child))
```
instead of chaining 2 `.filter()` operations.  Not sure which is best.